finder.py

```diff
@@ -231,31 +240,28 @@ def find_match(slug):
 	try:
 		found_matches = requests.get(url).json()
 	except json.decoder.JSONDecodeError as e:
-		raise OpendotaApiException from e
-
-	best_match = None
-	for match in found_matches:
-		if match["start_time"] < timestamp:
-			if best_match is None or match["start_time"] > best_match["start_time"]:
-				best_match = match
-	if best_match is None:
+		raise OpendotaApiException() from e
+
+	# Find the latest match that's earlier than `timestamp`.
+	start_time = itertools.itemgetter("start_time")
+	try:
+		best_match = max(
+			filter(found_matches, lambda match: start_time(match) < timestamp),
+			key=start_time)
+	except ValueError:
```

Comment:
I think the way it was done before is actually more clear for this. It also has the added bonus of not having to work with exceptions